File: functions/libsodium/variants/c/4_fBlaMka_1.c
#include <stdint.h>

static inline uint64_t fBlaMka(uint64_t x, uint64_t y)
{
    const uint64_t mask = UINT64_C(0xFFFFFFFF);
    return x + y + 2 * ((x & mask) * (y & mask));
}


int main(){
    fBlaMka(0,0);
}

File: functions/libsodium/variants/go/4_fBlaMka_1.go
package main

func fBlaMka(x, y uint64) uint64 {
	const m uint64 = 0xFFFFFFFF
	xy := x & m
	xy *= y & m
	return x + y + (xy << 1)
}


func main(){
    fBlaMka(0,0)
}
